framework/src/meshgenerators/TransitionLayerConnector.C

```diff
@@ -0,0 +1,205 @@
+
+  Real max_input_mesh_1_node_radius;
+  unsigned short invalid_boundary_type_1;
+  Real max_input_mesh_2_node_radius;
+  unsigned short invalid_boundary_type_2;
+  std::vector<dof_id_type> boundary_1_ordered_nodes;
+  std::vector<dof_id_type> boundary_2_ordered_nodes;
+
+  TransitionLayerTools::isBoundaryValid(*input_mesh_1,
+                                        max_input_mesh_1_node_radius,
+                                        invalid_boundary_type_1,
+                                        boundary_1_ordered_nodes,
+                                        _mesh_1_shift,
+                                        input_mesh_1_external_bids.front());
+  TransitionLayerTools::isBoundaryValid(*input_mesh_2,
+                                        max_input_mesh_2_node_radius,
+                                        invalid_boundary_type_2,
+                                        boundary_2_ordered_nodes,
+                                        _mesh_2_shift,
+                                        input_mesh_2_external_bids.front());
+
+  if (invalid_boundary_type_1 != 2)
+    paramError("boundary_1", "The provided boundary is not a single-segment boundary.");
+  if (invalid_boundary_type_2 != 2)
+    paramError("boundary_2", "The provided boundary is not a single-segment boundary.");
+
+  std::vector<Point> positions_vector_1;
+  std::vector<Point> positions_vector_2;
+
+  for (auto & boundary_1_node_id : boundary_1_ordered_nodes)
+    positions_vector_1.push_back(*input_mesh_1->node_ptr(boundary_1_node_id));
+
+  for (auto & boundary_2_node_id : boundary_2_ordered_nodes)
+    positions_vector_2.push_back(*input_mesh_2->node_ptr(boundary_2_node_id));
+
+  auto mesh = buildReplicatedMesh(2);
+  TransitionLayerTools::transitionLayerGenerator(*mesh,
+                                                 positions_vector_2,
+                                                 positions_vector_1,
+                                                 _num_layers,
+                                                 _block_id,
+                                                 _input_boundary_1_id,
+                                                 _input_boundary_2_id,
+                                                 _begin_side_boundary_id,
+                                                 _end_side_boundary_id,
+                                                 _type,
+                                                 _name,
+                                                 _use_quad_elements,
+                                                 _bias_parameter,
+                                                 _sigma);
+
+  if (_keep_inputs)
+  {
+    mesh->prepare_for_use();
+    mesh->stitch_meshes(
+        *input_mesh_1, _input_boundary_1_id, input_mesh_1_external_bids.front(), TOLERANCE, true);
+    mesh->stitch_meshes(
+        *input_mesh_2, _input_boundary_2_id, input_mesh_2_external_bids.front(), TOLERANCE, true);
```

Comment:
```
void libMesh::[ReplicatedMesh](https://libmesh.github.io/doxygen/classlibMesh_1_1ReplicatedMesh.html)::stitch_meshes	(	const ReplicatedMesh & 	other_mesh,
[boundary_id_type](https://libmesh.github.io/doxygen/namespacelibMesh.html#a3f4d8a52b5ef6ff6156ca209e12b5564) 	this_mesh_boundary,
[boundary_id_type](https://libmesh.github.io/doxygen/namespacelibMesh.html#a3f4d8a52b5ef6ff6156ca209e12b5564) 	other_mesh_boundary,
[Real](https://libmesh.github.io/doxygen/namespacelibMesh.html#aef28359fb4851abec08fc99cfbac118d) 	tol = [TOLERANCE](https://libmesh.github.io/doxygen/namespacelibMesh.html#ac935f799caf032c4008c5aa8facebaae),
bool 	clear_stitched_boundary_ids = false,
bool 	verbose = true,
bool 	use_binary_search = true,
bool 	enforce_all_nodes_match_on_boundaries = false 
)
```
I think you want to pass true to the last one too?

Reply:
Will do.